### DISCUSSION TITLE
Collection sort titles and pre-pended numbers like |2| and |7|

### DISCUSSION THREAD

**Discussion by gideunz** [CodyBerenson/PGMA-Modernized#266] [Q&A]
Hello!

Thanks for everyone's help on my previous post. I've got thing up-and-running, and while I've struggled a bit with some of the idiosyncrasies of the sites, I've finally wrestled my library into something like organized.

However, I am having a lot of trouble managing the collections. Here's what happened:

- Last week, I was going out of my mind trying to create the collections correctly, and I was convinced that my restarting and folder switching and futzing around with settings was creating multiple versions of the collections. This was only way I could explain why |2| or |7| or the like were pre-prended to the sort titles of the collections. I assumed I'd created so many versions that I was now on the 2nd or 7th identitical collection; I thought it might be like windows just puts (1) at the end of a file that has the same name as another file in the same folder.
- <img src="https://user-images.githubusercontent.com/62354877/256693778-b9c931b2-804c-43b9-8ed0-ac8ce8118b4d.png" width=50% height=50%>
- The pre-pendended numbers made sorting of the collections impossible, because nothing would sort alphabetically. (More of that below.) I even tried to do a find and replace in DB Browser for SQLite -- to get rid of a few hundred "|7|" at the beginning of sort-titles -- but that was an epic fail because of the way the collection variables are created. (I did learn about doing other things with DB Browser, so that's something.
- Finally, I ended up editing all of the sort titles manually. This took a very long time. A bunch of them refused to be edited. I deleted the ones that were uneditable. But then I matched some videos with, and new collections with pre-prended numbers showing up. The only way I'd know one of the collections had been screwed with is by scrolled past Z to where |2| and |7| ended up.  
- A coupled days ago, while search for information on this problem, I discovered a reference to the pre-pended numbers in the categories: https://github.com/CodyBerenson/PGMA-Modernized/issues/218#issuecomment-1402791264. So, these pre-pended numbers are for sorting the collections in the background, with |1| for System - including Agents, |2| for genres, and eventually |7| for actors. It was copies of collections, this was all deliberate.

So, my question is why are the collections sorted in this way? And Is there a way to stop this? I don't think it's a very intuitive way of handling collections for PGMA, mostly because Plex only allows to them to be sorted alphabetically, and the PGMA naming convention of the sort titles forces Plex to do something the user isnt expecting. I think it would make more sense to have the collections listed sort of like an an index with the specific design of the collection images (the ones for genres, countries, etc.) to work as signposts. 

Since I already had a lot of collections before I started using PGMA, the addition of the new ones with the pre-pended numbers didn't seem to have any rhyme or reason; if I had no collections and they all showed up in the PGMA order, I would have noticed there was a clear patten. But getting info on the pattern should be easier. There's nothing in the Usage Guide about it, so I think that would be great addition to the next version or however that would be handled. Also, if you want to make your own collections, not just rely on automated ones, you'd need to know sorting scheme to make it sensible. 

But for the time being, I have turned off collection creation for all of the agents, because whenever new collections are created, they get lost in the alphabetical order. And the agents also rewrite the sort titles some of my existing collections, which is making me a bit nues. I think I've deleted that |7| from my Jack Dixon collection five or sex times now.

Sorry for the length of this. Hope I didn't offend! I really got lost in the weeds on this, so I felt I needed to point this stuff out.

**Comment by JPH71** [comment]
Collections are sorted alphabetically within the various groups...
|0|....on iafd or not, stacked or not, compilations and iafd errors...
these are only created if you choose to create system collections

|1|. Agent scraped collections.. eg GEVI, AEBN...

|2| Genres

and so forth until

|7| cast...

I will look and see why these |x| are prepended to the title... and sort it
out...

Thanks for highlighting this...

I think that the iafd 403 error is sorted out... and I have worked on the
collections sorting...

As soon as I have finished testing.. I will forward to Cody..

If you had enabled system collections you could have then easily gone to
the error collection and rescraped them.

I will write a user guide so that Cody can add to the notes on this

Cheers

Jason
